### ISSUE TITLE
Support binaural for Cavernize's Render target

### ISSUE THREAD

**Issue by bluewidy** [VoidXH/Cavern#40]
With an ADM BWF file, it is possible to output binaural audio through DaVinci Resolve.
For the EAC3-JOC demo content provided directly by Dolby, the ADM BWF file works successfully. On the other hand I've tried creating ADM BWF files for several streaming Atmos content that provides EAC3-JOC, but all of them don't work for DaVinci Resolve.
<img width="257" alt="failed" src="https://user-images.githubusercontent.com/40682277/197374205-015c89a5-c944-414f-b064-ec080aec7e59.png">

So I am writing this post. Can I expect something like this?
![feature requeset](https://user-images.githubusercontent.com/40682277/197374410-bb057f48-7dcb-43f3-a089-6a0e2c0204af.png)

and it would be nice if .eac3 was added to the supported extension.
![feature requeset2](https://user-images.githubusercontent.com/40682277/197374709-267dc32f-8025-45d8-97b8-14f37380f1a4.png)

**Comment by ValZapod** [comment]
>Can I expect something like this?

That is 2.0 channels option for headphines renderer.

**Comment by bluewidy** [comment]
> That is 2.0 channels option

2.0 channel option? Where is it?
<img width="419" alt="Render target" src="https://user-images.githubusercontent.com/40682277/197376098-6b66b350-3f96-4139-b724-e2e9b549e12d.png">

**Comment by ValZapod** [comment]
Yes.

**Comment by ValZapod** [comment]
Just use this for now: 
![image](https://user-images.githubusercontent.com/31514790/197379044-7373248f-505c-4045-ba04-d9edff97a540.png)

**Comment by bluewidy** [comment]
You are actively trying to help me. Thank you very much. But as mentioned earlier, it doesn't work. 
Please try to convert this audio stream to an ADM BWF file and import it into DaVinci Resolve.
https://drive.google.com/file/d/148wUU-Utwe_owBX7B73qJuNcu3Zs6p4e/view?usp=sharing
Then you will see an error like this:
<img width="245" alt="import error" src="https://user-images.githubusercontent.com/40682277/197380633-424c0a0b-54fc-4100-90e1-238e52837f72.png">

**Comment by ValZapod** [comment]
Dolby Media encoder says invalid number of channels on the decoded ADM. So yeah.

**Comment by VoidXH** [comment]
Virtualizer render target added in 9876b95bd926f1cfe73f8980b919454740b927ec, .ec3 is also in the extension list.

**Comment by bluewidy** [comment]
Thank you so much for your hard work. I'll check it out after work.
> .ec3 is also in the extension list.
 Can you possibly add the .eac3 extension as well?

**Comment by VoidXH** [comment]
Done in e349658586b815ca8e05cf2c21e2b207cb2a5265, will be part of the next release.

**Comment by bluewidy** [comment]
Thank you sir!

**Comment by ValZapod** [comment]
```
ERROR: AtmosInfoAppWrapper: ADM BWF validation failed ATMOS_STORAGE_RES_SUCCESS
ERROR: Block format AB_00031007_00000003 must have rtime plus duration equal to the program duration
ERROR: Block format AB_00031008_00000003 must have rtime plus duration equal to the program duration
ERROR: Block format AB_00031009_00000039 must have rtime plus duration equal to the program duration
ERROR: Block format AB_0003100a_00000024 must have rtime plus duration equal to the program duration
ERROR: Block format AB_0003100b_00000024 must have rtime plus duration equal to the program duration
ERROR: Block format AB_0003100c_0000002c must have rtime plus duration equal to the program duration
ERROR: Block format AB_0003100d_00000012 must have rtime plus duration equal to the program duration
ERROR: Block format AB_0003100e_00000004 must have rtime plus duration equal to the program duration
ERROR: Block format AB_0003100f_00000026 must have rtime plus duration equal to the program duration
ERROR: Block format AB_00031010_00000012 must have rtime plus duration equal to the program duration.

```

and Yello does not work anymore.

**Comment by ValZapod** [comment]
Please revert d4063ca43e776891fa351d71a5e4deb53b08071a it breaks everything.

**Comment by bluewidy** [comment]
wow... you're still working...  I'm respectful. Do you live without sleep?

**Comment by ValZapod** [comment]
Yello is still broken, was good before https://github.com/VoidXH/Cavern/commit/d4063ca43e776891fa351d71a5e4deb53b08071a. atmos trak sample.ac3 works now. But not star wars, that never worked.

Star wars:
[hncthjx.zip](https://github.com/VoidXH/Cavern/files/9858880/hncthjx.zip)

Yello:

[01 - Yello - Waba Duba.zip](https://github.com/VoidXH/Cavern/files/9858889/01.-.Yello.-.Waba.Duba.zip)

**Comment by VoidXH** [comment]
On it.

**Comment by bluewidy** [comment]
> atmos trak sample.ac3 works now.

https://drive.google.com/file/d/148wUU-Utwe_owBX7B73qJuNcu3Zs6p4e/view?usp=sharing
It is true that this sample works. However, the samples I've submitted in the past still don't work.

> Here is a sample file for not working timezones. I have succeeded in narrowing the scope with super precision! https://drive.google.com/file/d/1eGIet8kp0qBYiUYfuaT7RNsD4QaVRv0S/view?usp=sharing

**Comment by ValZapod** [comment]
>However, the samples I've submitted in the past still don't work

That is the same issue, invalid number of channels.

**Comment by VoidXH** [comment]
Star Wars and atmos trak sample works now, Yello and Not working timeline sample is yet to be fixed.

**Comment by ValZapod** [comment]
Yello sample does work if you get to before https://github.com/VoidXH/Cavern/commit/d4063ca43e776891fa351d71a5e4deb53b08071a

Unless you see some issue with how it worked back then?

**Comment by VoidXH** [comment]
I wrote a validator and manually checked the files, including "Not working timeline sample". It's a great help as it's only a single second, but I can't see anything. Do any of your tools give any feedback other than "shit's broke yo"?

**Comment by ValZapod** [comment]
>Do any of your tools give any feedback other than "bad ADM"?

Nuendo says Yello sample is missing atmos at all. It does see automation though. But Atmos does not render. On master Yello is much smaller than on before https://github.com/VoidXH/Cavern/commit/d4063ca43e776891fa351d71a5e4deb53b08071a

Why cannot you just compare the files in Beyond Compare? There is no ANY difference in Mediainfo.

BTW, I did not check Yello yet your very newest changes. Will do.

**Comment by VoidXH** [comment]
Since timings were fixed, all blocks are different in Beyond Compare. Also, "Not working timeline sample" never worked, which was the case with Star Wars, but its issue was actually the number of digits for interpolation. This is also fixed now.



> Yello sample is missing atmos at all

What is the exact error message?

**Comment by ValZapod** [comment]
>Yello and Not working timeline sample is yet to be fixed.

You did not check them. Yello works.

**Comment by VoidXH** [comment]
> You did not check them. Yello works.

DACT still doesn't like it. Does the 1 second sample also work?

**Comment by ValZapod** [comment]
Not working timeline sample.eac3 is still broken.

**Comment by VoidXH** [comment]
Checked manually and there's 26 channels and 26 ADM objects. I have no idea what this issue is, but it's funny that if you convert a very short clip, it doesn't work, but when you test a longer version of the same, it does.

**Comment by ValZapod** [comment]
hncthjx.zip star wars does indeed work. Nice. Some other sample also works.

**Comment by ValZapod** [comment]
Does Dolby Atmos Conversion Tool work with yello before https://github.com/VoidXH/Cavern/commit/d4063ca43e776891fa351d71a5e4deb53b08071a? If it does not, it is an issue with it.

**Comment by VoidXH** [comment]
> Does Dolby Atmos Conversion Tool work with yello before [d4063ca](https://github.com/VoidXH/Cavern/commit/d4063ca43e776891fa351d71a5e4deb53b08071a)?

No, not even with the ones before.

**Comment by ValZapod** [comment]
>I don't know what these files are :D

Html files. Log files as used in Dolby Media encoder.

**Comment by bluewidy** [comment]
My question is... If I re-encode from EAC3-JOC to ADM BWF, then from ADM BWF to EAC3-JOC, does the quality degrade compared to the original?

**Comment by ValZapod** [comment]
>quality degrade compared to the original

"Generation loss." Yes. https://en.m.wikipedia.org/wiki/Generation_loss

**Comment by VoidXH** [comment]
> Html files. Log files as used in Dolby Media encoder.

But what should I check in them?

> it is only obvious if encoded to eac3.

Simple cutting of Atmos content also breaks priming if there was any.

> It was not.

See #6.

**Comment by bluewidy** [comment]
> re-encode from EAC3-JOC to ADM BWF

The grammar is wrong because I used Google Translate.

> "Generation loss." Yes.

The reason I was curious about this is because I had a lot of noise when I got the ADM BWF.

**Comment by ValZapod** [comment]
>The reason I was curious about this is because I had a lot of noise when I got the ADM BWF.

See https://github.com/VoidXH/Cavern/issues/17

**Comment by bluewidy** [comment]
The funny thing is, we're talking in a binaural topic place.

**Comment by bluewidy** [comment]
To put the news properly on the binaural topic, I fired the final flare at Dolby. 

 Currently, DRP's CLI supports 2.0, 3.1, 5.1 and 7.1 PCM export from DD+ and TrueHD audio streams. So I asked for support for Binural export as well, and yesterday I got a reply from a senior support engineer that he would forward my feedback to the program team. 

 But I don't know what percent chance they will accept my feedback.
